crates/data_components/src/iceberg/catalog.rs

```diff
@@ -0,0 +1,252 @@
+
+use async_trait::async_trait;
+use iceberg::{
+    table::Table, Catalog, Error as IcebergError, ErrorKind, Namespace, NamespaceIdent,
+    Result as IcebergResult, TableCommit, TableCreation, TableIdent,
+};
+use iceberg_catalog_rest::{RestCatalog as IcebergRestCatalog, RestCatalogConfig};
+
+#[derive(Debug)]
+pub struct RestCatalog {
+    inner: IcebergRestCatalog,
+}
+
+/// The kinds of tables that can be returned from the `RestCatalog`.
+#[derive(Debug, Clone, Copy)]
+pub enum IcebergTableKind {
+    Iceberg,
+    Spice,
+}
+
+/// The table type that can be returned from the `RestCatalog`.
+#[derive(Debug)]
+pub enum IcebergTable {
+    Iceberg(Table),
+    Spice(String),
+}
+
+impl RestCatalog {
+    #[must_use]
+    #[allow(clippy::missing_panics_doc)]
+    pub fn new(catalog_config: RestCatalogConfig, table_kind: IcebergTableKind) -> Self {
+        // This will be removed in the next PR.
+        assert!(
+            matches!(table_kind, IcebergTableKind::Iceberg),
+            "RestCatalog currently only supports Iceberg tables"
+        );
+        Self {
+            inner: IcebergRestCatalog::new(catalog_config),
+        }
+    }
+
+    pub async fn load_table(&self, table: &TableIdent) -> IcebergResult<IcebergTable> {
+        let table = self.inner.load_table(table).await?;
+        Ok(IcebergTable::Iceberg(table))
+    }
+}
+
+#[async_trait]
+impl Catalog for RestCatalog {
+    /// List namespaces inside the catalog.
+    async fn list_namespaces(
+        &self,
+        parent: Option<&NamespaceIdent>,
+    ) -> IcebergResult<Vec<NamespaceIdent>> {
+        self.inner.list_namespaces(parent).await
+    }
+
+    /// Create a new namespace inside the catalog.
+    async fn create_namespace(
+        &self,
+        _namespace: &NamespaceIdent,
+        _properties: HashMap<String, String>,
+    ) -> IcebergResult<Namespace> {
+        return Err(IcebergError::new(
+            ErrorKind::FeatureUnsupported,
+            "Not implemented",
+        ));
+    }
+
+    /// Get a namespace information from the catalog.
+    async fn get_namespace(&self, namespace: &NamespaceIdent) -> IcebergResult<Namespace> {
+        self.inner.get_namespace(namespace).await
+    }
+
+    /// Check if namespace exists in catalog.
+    async fn namespace_exists(&self, namespace: &NamespaceIdent) -> IcebergResult<bool> {
+        self.inner.namespace_exists(namespace).await
+    }
+
+    /// Update a namespace inside the catalog.
+    ///
+    /// # Behavior
+    ///
+    /// The properties must be the full set of namespace.
+    async fn update_namespace(
+        &self,
+        _namespace: &NamespaceIdent,
+        _properties: HashMap<String, String>,
+    ) -> IcebergResult<()> {
+        return Err(IcebergError::new(
+            ErrorKind::FeatureUnsupported,
+            "Not implemented",
+        ));
+    }
+
+    /// Drop a namespace from the catalog.
+    async fn drop_namespace(&self, _namespace: &NamespaceIdent) -> IcebergResult<()> {
```

Comment:
can a user trigger this, or will only developers see this? If user's see it, I'm worried about what this message looks like to a user.

Reply:
There is no codepath that triggers this, its part of the trait definition from the external `iceberg` crate. We only use the read-only methods